File: app/src/main/java/com/example/myapplication/data/mapper/UserRoomMapper.java
package com.example.myapplication.data.mapper;

import com.example.myapplication.data.local.room.UserRoomEntity;
import com.example.myapplication.data.remote.aidl.UserParcelable;
import com.example.myapplication.domain.model.User;

public final class UserRoomMapper {

    private UserRoomMapper() {
    }

    public static User toDomain(UserRoomEntity entity) {
        return new User(entity.id, entity.name, entity.age, entity.weight);
    }

    public static UserRoomEntity toRoomEntity(User user) {
        return new UserRoomEntity(
                user.getId(),
                user.getName(),
                user.getAge(),
                user.getWeight()
        );
    }

    public static UserParcelable toParcelable(UserRoomEntity entity) {
        return new UserParcelable(entity.id, entity.name, entity.age, entity.weight);
    }

    public static UserRoomEntity toRoomEntity(UserParcelable userParcelable) {
        return new UserRoomEntity(
                userParcelable.getId(),
                userParcelable.getName(),
                userParcelable.getAge(),
                userParcelable.getWeight()
        );
    }
}

File: app/src/main/java/com/example/myapplication/data/mapper/UserRoomMapper.kt
package com.example.myapplication.data.mapper

import com.example.myapplication.data.local.room.UserRoomEntity
import com.example.myapplication.data.remote.aidl.UserParcelable
import com.example.myapplication.domain.model.User

fun UserRoomEntity.toDomain(): User = User(
    id = id,
    name = name,
    age = age,
    weight = weight,
)

fun User.toRoomEntity(): UserRoomEntity = UserRoomEntity(
    id,
    name,
    age,
    weight,
)

fun UserRoomEntity.toParcelable(): UserParcelable = UserParcelable(
    id = id,
    name = name,
    age = age,
    weight = weight,
)

fun UserParcelable.toRoomEntity(): UserRoomEntity = UserRoomEntity(
    id,
    name,
    age,
    weight,
)
